packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/mutation.ts

```diff
@@ -0,0 +1,110 @@
+import { compoundExpression, list, methodCall, obj, printBlock, qref, ref, set, str } from 'graphql-mapping-template';
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { FieldDefinitionNode, ObjectTypeDefinitionNode } from 'graphql';
+import { ConfiguredAuthProviders, RoleDefinition } from '../../../utils';
+import { constructAuthorizedInputStatement, generateAuthRulesFromRoles, validateAuthResult } from './common';
+
+export const generateAuthExpressionForCreate = (
+  ctx: TransformerContextProvider,
+  providers: ConfiguredAuthProviders,
+  roles: Array<RoleDefinition>,
+  fields: ReadonlyArray<FieldDefinitionNode>,
+): string => {
+  const expressions = [];
+  const operation = 'create';
+  expressions.push(compoundExpression(generateAuthRulesFromRoles(roles, fields)));
+  expressions.push(
+    set(
+      ref('authResult'),
+      methodCall(
+        ref('util.authRules.mutationAuth'),
+        ref('authRules'),
+        str(operation),
+        ref('ctx.args.input'),
+      ),
+    ),
+  );
+  expressions.push(
+    validateAuthResult(),
+    constructAuthorizedInputStatement('ctx.args.input'),
+  );
+  return printBlock('Authorization rules')(compoundExpression(expressions));
+};
+
+export const generateAuthExpressionForUpdate = (
+  providers: ConfiguredAuthProviders,
+  roles: Array<RoleDefinition>,
+  fields: ReadonlyArray<FieldDefinitionNode>,
+): string => {
+  const expressions = [];
+  const operation = 'update';
+  expressions.push(compoundExpression(generateAuthRulesFromRoles(roles, fields)));
+  expressions.push(
+    set(
+      ref('authResult'),
+      methodCall(
+        ref('util.authRules.mutationAuth'),
+        ref('authRules'),
+        str(operation),
+        ref('ctx.args.input'),
+        ref('ctx.source'),
+      ),
+    ),
+  );
+  expressions.push(validateAuthResult());
+  return printBlock('Authorization rules')(compoundExpression(expressions));
+};
+
+export const generateAuthExpressionForDelete = (
+  providers: ConfiguredAuthProviders,
+  roles: Array<RoleDefinition>,
+  fields: ReadonlyArray<FieldDefinitionNode>,
+): string => {
+  const expressions = [];
+  const operation = 'delete';
+  expressions.push(compoundExpression(generateAuthRulesFromRoles(roles, fields)));
+  expressions.push(
+    set(
+      ref('authResult'),
+      methodCall(
+        ref('util.authRules.mutationAuth'),
+        ref('authRules'),
+        str(operation),
+        ref('ctx.args.input'),
+        ref('ctx.source'),
+      ),
+    ),
+  );
+  expressions.push(validateAuthResult());
+  return printBlock('Authorization rules')(compoundExpression(expressions));
+};
```

Comment:
The above 3 methods `generateAuthExpressionFor...` has significant parts in common, can centralize these in a single method.

Reply:
This has been refactored into a helper method.